### PR TITLE
Fix various typos

### PR DESCRIPTION
**Pull request by luzpaz**
Typos found via `codespell -q 3 -S ./vendor`

### CODE REVIEW

**Comment by hodoulp** [comment]
Hi @luzpaz,

Thanks for your contribution on the OpenColorIO project. As you might have noticed the documentation still needs some love.  In order to help us you need to sign the CLA so that your pull request could be authorized.

**Comment by luzpaz** [comment]
@hodoulp signed CLA :+1:

File: tests/cpu/OpOptimizers_tests.cpp [hunk 1]
```diff
@@ -490,8 +490,8 @@ OCIO_ADD_TEST(OpOptimizers, optimization2)
     // 3 Range, bit depth conversion identity
     // 4 Matrix, almost identity
     // 5 Range, clamp identity
-    // 6 Lut1D, half domain, raw halfs, identity
-    // 7 Lut1D, raw halfs, identity
+    // 6 Lut1D, half domain, raw halves, identity
```
Comment (hodoulp) [inline tests/cpu/OpOptimizers_tests.cpp]: Please revert to `half`

File: tests/cpu/OpOptimizers_tests.cpp [hunk 2]
```diff
@@ -490,8 +490,8 @@ OCIO_ADD_TEST(OpOptimizers, optimization2)
...
+    // 7 Lut1D, raw halves, identity
```
Comment (hodoulp) [inline tests/cpu/OpOptimizers_tests.cpp]: Please revert to `half`

File: tests/cpu/ops/lut1d/Lut1DOpCPU_tests.cpp [hunk 3]
```diff
@@ -2440,8 +2440,8 @@ OCIO_ADD_TEST(Lut1DRenderer, lut_1d_inv_half_fclut)
 
     // TODO: Review the test to add LUT & inverse LUT together when optimization is reworked.
 
-    // Lustre log_default.fclut.  All positive halfs map to unique 16-bit ints
-    // so it's a good test to see that the inverse can restore the halfs losslessly.
+    // Lustre log_default.fclut.  All positive halves map to unique 16-bit ints
```
Comment (hodoulp) [inline tests/cpu/ops/lut1d/Lut1DOpCPU_tests.cpp]: Please revert to `halfs`

File: tests/cpu/ops/lut1d/Lut1DOpCPU_tests.cpp [hunk 4]
```diff
@@ -2440,8 +2440,8 @@ OCIO_ADD_TEST(Lut1DRenderer, lut_1d_inv_half_fclut)
...
+    // so it's a good test to see that the inverse can restore the halves losslessly.
```
Comment (hodoulp) [inline tests/cpu/ops/lut1d/Lut1DOpCPU_tests.cpp]: Please revert to `halfs`

File: tests/cpu/ops/lut1d/Lut1DOpCPU_tests.cpp [hunk 5]
```diff
@@ -2453,7 +2453,7 @@ OCIO_ADD_TEST(Lut1DRenderer, lut_1d_inv_half_fclut)
                                                                OCIO::BIT_DEPTH_F32,
                                                                DefaultNoLutInvFast));
 
-    // Test all positive halfs (less than inf) round-trip losslessly.
+    // Test all positive halves (less than inf) round-trip losslessly.
```
Comment (hodoulp) [inline tests/cpu/ops/lut1d/Lut1DOpCPU_tests.cpp]: Please revert to `halfs`

File: tests/cpu/OpOptimizers_tests.cpp (hunk 1 above)
Comment (lgritz) [reply tests/cpu/OpOptimizers_tests.cpp]: Agreed. This refers to the plural of the data type `half`, not the English word "halves" meaning 1/2 of something.

**Comment by luzpaz** [comment]
Reverted requested commits in b636b2c

**Comment by luzpaz** [comment]
@hodoulp sorry to disturb, but I found a few more typos in b56e67a (do you mind reviewing them?)

**Pull request merged**